ir/src/symbol_table.rs

```diff
@@ -132,13 +132,18 @@ impl SymbolTable {
     pub(super) fn insert_trace_columns(
         &mut self,
         trace_segment: TraceSegment,
-        columns: &[Identifier],
+        columns: &[TraceCol],
     ) -> Result<(), SemanticError> {
         self.set_num_trace_segments(trace_segment);
 
-        for (idx, Identifier(name)) in columns.iter().enumerate() {
-            let trace_column = TraceColumn::new(trace_segment, idx);
-            self.insert_symbol(name, IdentifierType::TraceColumn(trace_column))?;
+        for (idx, trace_col) in columns.iter().enumerate() {
+            match trace_col {
+                TraceCol::Single(Identifier(name)) => {
+                    let trace_column = TraceColumn::new(trace_segment, idx);
+                    self.insert_symbol(name, IdentifierType::TraceColumn(trace_column))?;
+                }
+                TraceCol::Group(_, _) => todo!(),
+            }
```

Comment:
I think it will end up being cleaner here, in the symbol table, and elsewhere in the IR if we don't have to deal with multiple variants like this. See my comments in the AST.